src/Common/src/System/Data/Common/DbConnectionOptions.Common.cs

```diff
@@ -407,6 +407,7 @@ private static bool IsValueValidInternal(string keyvalue)
                 bool compValue = s_connectionStringValidValueRegex.IsMatch(keyvalue);
                 Debug.Assert((-1 == keyvalue.IndexOf('\u0000')) == compValue, "IsValueValid mismatch with regex");
 #endif
+                // string.Contains(char) is .NetCore2.1+ specific
```

Comment:
There's a compilation error if someone tries it, right? (Or does it fall back to Linq?). If it's an error, I don't think these comments are worthwhile. There are plenty of other cases where such code avoids calling API that isn't available earlier.

Reply:
@danmosemsft we did a sweep of the whole codebase to update the calls. That comment is a sentinel to locate those calls again should NetFx catch up. You are right they're not useful in `Assert`s though.